site_scons/buildutils.py

```diff
@@ -34,6 +36,373 @@
     TPathLike = TypeVar("TPathLike", Path, str)
 
 
+class Option:
+    """Object corresponding to SCons configuration option"""
+
+    def __init__(self,
+            name: str,
+            description: str,
+            default: Union[bool, str],
+            choices: list=None):
+        self.name = name
+        self.description = Option._deblank(description)
+        self.default = default
+        self.choices = choices
+
+        self._wrapper = textwrap.TextWrapper(width=80)
+        self.set_wrapper_indent(4)
+
+    @property
+    def parameters(self) -> tuple:
+        """Parameters of the SCons configuration option"""
+        if self.choices is None:
+            return self.name, self.description, self.default
+        return self.name, self.description, self.default, self.choices
+
+    def to_scons(self, env: "SCEnvironment"=None) -> "SCVariables":
```

Comment:
When type information is included, there's usually spaces around `=`, even in the function definition. This should also be optional:
```suggestion
    def to_scons(self, env: "Optional[SCEnvironment]" = None) -> "SCVariables":
```

Reply:
This one I'm actually not sure about. While I added the spaces around `=` throughout, the `Optional[...]` appears to add clutter: from the context, it is clear that this is optional already, as default values are being specified?

Reply:
See my other comment, sorry I didn't make that comment here

Reply:
Alright. Replaced by `"Union[SCEnvironment, None]"`